s3/pkg/service/bucket.go

```diff
@@ -210,13 +231,38 @@ func (s *s3Service) DeleteBucket(ctx context.Context, in *pb.Bucket, out *pb.Bas
 		err = ErrBucketNotEmpty
 		return nil
 	}
+
+	// We need to find the backend in which bucket will be created
+	backend, err := utils.GetBackend(ctx, s.backendClient, bucket.DefaultLocation)
+	if err != nil {
+		log.Errorln("failed to get backend client with err:", err)
+		return err
+	}
+
+	log.Info("the backend is:\n", backend)
+
+	sd, err := driver.CreateStorageDriver(backend.Type, backend)
+	if err != nil {
+		log.Errorln("failed to create storage. err:", err)
+		return err
+	}
+
+	log.Info("The driver for the backend is:\n", sd)
+
+	err = sd.BucketDelete(ctx, in)
+	log.Error("The error while deleting bucket:", err)
```

Comment:
nit: redundant error log .
will be there in success scenario also

Reply:
done. Thanks